edit/harvesterhci.io.virtualmachineimage.vue

```diff
@@ -0,0 +1,268 @@
+<script>
+import CruResource from '@/components/CruResource';
+import Tabbed from '@/components/Tabbed';
+import Tab from '@/components/Tabbed/Tab';
+import LabeledInput from '@/components/form/LabeledInput';
+import KeyValue from '@/components/form/KeyValue';
+import NameNsDescription from '@/components/form/NameNsDescription';
+import RadioGroup from '@/components/form/RadioGroup';
+import LabelValue from '@/components/LabelValue';
+import CreateEditView from '@/mixins/create-edit-view';
+import { _EDIT } from '@/config/query-params';
+import { IMAGE_FILE_FORMAT } from '@/config/constant';
+import { HCI as HCI_ANNOTATIONS } from '@/config/labels-annotations';
+import { exceptionToErrorsArray } from '@/utils/error';
+
+const DOWNLOAD = 'download';
+const UPLOAD = 'upload';
+
+export default {
+  name: 'EditImage',
+
+  components: {
+    Tab,
+    Tabbed,
+    KeyValue,
+    CruResource,
+    LabeledInput,
+    NameNsDescription,
+    RadioGroup,
+    LabelValue,
+  },
+
+  mixins: [CreateEditView],
+
+  props: {
+    value: {
+      type:     Object,
+      required: true,
+    },
+  },
+
+  data() {
+    if ( !this.value.spec ) {
+      this.$set(this.value, 'spec', { sourceType: DOWNLOAD });
+    }
+
+    return {
+      url:         this.value.spec.url,
+      files:       [],
+      displayName: '',
+      resource:    '',
+      headers:     {},
+      fileUrl:     '',
+      file:        '',
+    };
+  },
+
+  computed: {
+    uploadFileName() {
+      return this.file?.name || '';
+    },
+
+    isEdit() {
```

Comment:
This already exists in the inherited mixin `CreateEditView`

Reply:
I will do it